llvm/test/CodeGen/X86/select-phi-s16-fp.ll

```diff
@@ -0,0 +1,110 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -global-isel=0 -mcpu=generic -mtriple=x86_64-apple-darwin %s -o - | FileCheck %s
+
+; For all these tests we disable optimizations through function attributes
+; because the code we are exercising here needs phis and we want to keep the
+; IR small.
+
+; This code used to crash in SDISel because f16 was promoted to f32 through
+; a `f32 = vector_extract_elt <1 x f16>, i32 0`, which is illegal.
+; The invalid SDNode and thus, the crash was only exposed by the constant
+; folding.
+define void @phi_vec1half_to_f32_with_const_folding(ptr %dst) #0 {
```

Comment:
`half` tests pass on X86 because it's a legal type. You can only leave `bfloat` tests.

Reply:
no reason to not test both

Reply:
I'm ok with keeping these tests, but at least the comments should move to phi_vec1bf16_to_f32 and replace with bf16 since they are not accurate here.

Reply:
Hmm, I believe the f16 lowering was crashing too. Let me double check.

Reply:
@phoebewang you're right, it doesn't crash. I could have sworn it was.
Anyhow, I'll update the comments.